### DISCUSSION TITLE
Problems with SmartFormComponent and MUI - Mixing elements w/ Controller and w/o Controller

### DISCUSSION THREAD

**Discussion by keebeegee** [react-hook-form/react-hook-form#2402] [Q&A] [answered]
Hi folks,

I am running into an issue implementing [SmartFormComponent](https://react-hook-form.com/advanced-usage#SmartFormComponent). I assume that this is the "best practice" approach to build a form out of reusable form components/widgets. 

Problem is that using the SmartFormComponent approach the MUI widget will not be registered with the Form. Submitting the form does not show the value of the MUI element: https://codesandbox.io/s/stoic-euler-69jtw

Thanks in advance for your help. 

Cheers - kbg

**Comment by bluebill1049** [comment, marked as answer]
it's not the best approach, it's one of the approaches :) for MUI you probably should use controller.

**Reply by keebeegee** [reply]
I actually do use a controller for the the MUI element - see https://codesandbox.io/s/stoic-euler-69jtw
If the "SmartFormComponent" way is not the best approach, what are the alternatives? 
I tried several ways composing a form using MUI elements that I import from other places (DRY) with no luck.

**Reply by keebeegee** [reply]
Here's the solution - ALL form elements need to use the "Controller": 
https://codesandbox.io/s/ecstatic-bhaskara-u449p
